File: server/src/main/java/ua/com/alevel/service/educationalprocess/LessonService.java
package ua.com.alevel.service.educationalprocess;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.educationalprocess.Lesson;
import ua.com.alevel.persistence.repository.educationalprocess.LessonRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class LessonService extends AbstractService<Lesson, LessonRepository> {

    private final LessonRepository lessonRepository;
    private final CrudRepositoryHelper<Lesson, LessonRepository> crudRepositoryHelper;

    public LessonService(LessonRepository lessonRepository, CrudRepositoryHelper<Lesson, LessonRepository> crudRepositoryHelper) {
        super(lessonRepository, crudRepositoryHelper);
        this.lessonRepository = lessonRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}

File: client-tutor/src/app/shared/services/educationalprocess/lesson.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {Lesson} from "../../models/educationalprocess/lesson.model";

@Injectable({
  providedIn: 'root'
})
export class LessonService extends AbstractService<Lesson> {

  override resourceUrl = '/api/lessons';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}
